powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/repository/SignatureAuditRepository.java

```diff
@@ -38,7 +39,8 @@
      * @param endingDate   Ending date (date "to").
      * @return List of {@link SignatureEntity} instances.
      */
-    List<SignatureEntity> findByActivation_UserIdAndTimestampCreatedBetweenOrderByTimestampCreatedDescIdDesc(String userId, Date startingDate, Date endingDate);
+    @Query("SELECT s FROM SignatureEntity s WHERE s.activation.userId = ?1 AND s.timestampCreated BETWEEN ?2 AND ?3 ORDER BY s.timestampCreated DESC")
```

Comment:
The `ORDER BY` clause is missing `AND id DESC`, to keep the query equivalent and assure best attempt ordering for records with the same timestamp.

Reply:
Good catch!